README.md

```diff
@@ -51,6 +51,48 @@ from the `global-options`. Likewise, the `role` key is optional and defaults to
 if omitted. Valid roles are "client", "server" and "all".
 
 
+## Requirements file
+The requirements file defines all the validation and transformation parameters for a
+specific benchmark. The file contains three main blocks: `default`, `mandatory` and
+`validation`. The example below is a simplified version of fio benchmark requirements
+file:
+```
+{
+    "default": {
+        "rw": ["read", "randread"],
+        "bs": ["16k"]
+    },
+    "mandatory": {
+        "write_hist_log": ["fio"],
+        "log_hist_msec": ["10000"]
+    },
+    "validation": {
+        "size_KMG" : {
+            "description" : "bytes in k/K (1024), m/M (1024^2) or g/G (1024^3): 4k 16M 1g",
+            "arguments" : [ "param1", "filesize", "io_size", "mem" ],
+            "value" : "[0-9]+[kbmgKBMG]",
+            "transform" : [ "s/([0-9]+)[gG]/($1*1024).\"M\"/e",
+                            "s/([0-9]+)[mM]/($1*1024).\"K\"/e"
+            ]
+        }
+    }
+}
+```
+
+#### default
+Defines all the default param values. Multiplex assumes these default params
+if the param is not present in the mv-paramns.json input file.
+
+#### mandatory
+Defines all the mandatory params. Multiplex checks if all the mandatory params
+are specified in the mv-params.json input file and it fails otherwise.
```

Comment:
I think we want the default behavior to be that multiplex uses the mandatory parameters from the requirements file unless those parameters are explicitly listed in the mv-params.json

Reply:
wait... isn't it the default ones to be used if not specified in mv-params.json ?

The mandatory params should be for validating params that need to be present.

Or maybe we combine them into one: "default", adding the default values for mandatory ones  ?

Reply:
So I think we are mixing/confusing terms here.

The `default` set of parameters are supposed to be what is used if no parameters are supplied by the user.

The `mandatory` parameters are the basic parameters that our harness (ie. `crucible` + `rickshaw` + `bench-*`) need to function properly -- our post processing may require certain options be enabled so that certain data is available (ie. a latency histogram for example).  So the `mandatory` parameters should always be appended to the list of parameters to use to guarantee basic functionality of the harness.  That said, there may be times when the user needs to experiment with these parameters for various reasons so the ability to override the `mandatory` parameters is a necessary evil.

Reply:
We also have discussed the concept of parameter `presets`.  This would allow the `bench-*` project to define a list of pre-defined parameter sets to allow the user to easily run a variety of tests.  In the case of fio, that could be something like this:

```
"presets": [
  {
    "name": "sequential-read",
    "params": [
       { "arg": "rw", "values": [ "read" ] },
       { "arg": "bs", "values": [ "4k" ] },
       ...
    ]
  },
  {
    "name": "sequential-write",
    "params": [
       { "arg": "rw", "values": [ "write" ] },
       { "arg": "bs", "values": [ "4k" ] },
       ...

    ]
  }
]
```

Reply:
added